sw/device/lib/dif/dif_uart.c

```diff
@@ -498,25 +496,19 @@ dif_uart_result_t dif_uart_fifo_reset(const dif_uart_t *uart,
       mmio_region_read32(uart->params.base_addr, UART_FIFO_CTRL_REG_OFFSET);
 
   if (reset == kDifUartFifoResetRx || reset == kDifUartFifoResetAll) {
-    reg = bitfield_field32_write(reg,
-                                 (bitfield_field32_t){
-                                     .mask = 0x1,
-                                     .index = UART_FIFO_CTRL_RXRST_BIT,
-                                 },
-                                 1);
+    reg = bitfield_bit32_write(reg, UART_FIFO_CTRL_RXRST_BIT, true);
+    mmio_region_write32(uart->params.base_addr, UART_FIFO_CTRL_REG_OFFSET, reg);
+    reg = bitfield_bit32_write(reg, UART_FIFO_CTRL_RXRST_BIT, false);
+    mmio_region_write32(uart->params.base_addr, UART_FIFO_CTRL_REG_OFFSET, reg);
   }
 
   if (reset == kDifUartFifoResetTx || reset == kDifUartFifoResetAll) {
-    reg = bitfield_field32_write(reg,
-                                 (bitfield_field32_t){
-                                     .mask = 0x1,
-                                     .index = UART_FIFO_CTRL_TXRST_BIT,
-                                 },
-                                 1);
+    reg = bitfield_bit32_write(reg, UART_FIFO_CTRL_TXRST_BIT, true);
+    mmio_region_write32(uart->params.base_addr, UART_FIFO_CTRL_REG_OFFSET, reg);
+    reg = bitfield_bit32_write(reg, UART_FIFO_CTRL_TXRST_BIT, false);
+    mmio_region_write32(uart->params.base_addr, UART_FIFO_CTRL_REG_OFFSET, reg);
```

Comment:
> nit: Do these have to be sequential? Resetting both like this would take 4 writes instead of 2.

Good point can do at the same time.

> Also, the requirement to write 0 after writing 1 is not documented in here.

It is something that @eunchan suggested in `HW_USAGE_REVIEWED`. @eunchan did I catch that correctly?

Reply:
I recall that we had to write a 1 then write a 0 from the meeting with @eunchan, but I'll let him clarify how it should be done.

Reply:
Yeah right. SW should write 1 then 0 to clear the reset signal. I will clarify it in the document.

Reply:
Do we need to do it this way? The verilog code now uses 
  assign uart_fifo_rxrst  = reg2hw.fifo_ctrl.rxrst.q & reg2hw.fifo_ctrl.rxrst.qe;
i.e. the hardware only does something based on the 1 being written not the 1 being sustained. 

So maybe we should set the register bit not to have a storage element, have the software only write a 1 to cause the reset and always read back zero? Which I thought would happen with swaccess: "wo", hwaccess: "hro" in the hjson but maybe we need hwext too.

This would remove the need for the explicit write back to zero (esp when the register will always read as zero). It would also remove the potential confusion caused because the fifo can start being filled while the bit is a 1 (the fifo is only actually reset during the cycle where the 1 is written not for the entire duration of the bit being 1).

(Aside: We may also want to change the rxilvl and txilvl bits to be hro to avoid the need to tie off d and de to zero in the verilog?)

(jumping in because I was just reviewing this as I factored all the changes into usbuart.)